cachi2/core/package_managers/gomod.py

```diff
@@ -664,6 +664,27 @@ def _get_repository_name(source_dir: RootedPath) -> str:
     return f"{url.hostname}{url.path.rstrip('/').removesuffix('.git')}"
 
 
+def _get_gomod_version(source_dir: RootedPath) -> Optional[str]:
+    """Return the required/recommended version of Go from go.mod.
+
+    We need to extract the desired version of Go ourselves as older versions of Go might fail
+    due to e.g. unknown keywords or unexpected format of the version (yes, Go always performs
+    validation of go.mod).
+
+    If we cannot extract a version from the 'go' line, we return None, leaving it up to the caller
+    to decide what to do next.
+    """
+    # Spawning Go as a process could be expensive, so instead of using this manual extraction only
+    # as a fallback we use it right away.
```

Comment:
This commentary is a leftover from earlier revisions that no longer applies, I'll drop it in the next revision when I'll likely have to respin.